isaric/parsers/rapid_observations.toml

```diff
@@ -0,0 +1,782 @@
+            { field = "hostdat", description = "Admission date at this facility" },
+            { field = "cestdat", description = "Onset date of first/earliest symptom" },
+            { field = "dsstdat", description = "Date of enrolment" }
+        ]
+
+[adtl.defs."followupDateHierarchy"]
+combinedType = "firstNonNull"
+fields = [
+{ field = "flw_survey_date", description = "About you and your COVID-19 illness (if you're completing this survey on behalf of a child or adult that you care for, all the questions relate to their health and wellbeing)" },
+{ field = "flw2_survey_date", description = "About you and your COVID-19 illness (if you're completing this survey on behalf of a child or adult that you care for, all the questions relate to their health and wellbeing)" },
+{ field = "daily_date", description = "Date of follow-up" },
+{ field = "dsstdtc", description = "Outcome date" }
+]
+
+[[observation]]
+name = "haemorrhage"
+date = { ref = "admissionDateHierarchy" }
+phase = "admission"
+is_present = { field = "bleed_ceoccur_v2", ref = "Y/N/NK" }
+if = { bleed_ceoccur_v2 = { "!=" = 3 } }
+occurrence_period = 
+#description = "Bleeding (Haemorrhage)"
+
+[[observation]]
+name = "cough"
+date = { ref = "admissionDateHierarchy" }
+phase = "admission"
+is_present = { if field = "cough_ceoccur_v2", ref = "Y/N/NK" }
+if = { cough_ceoccur_v2 = { "!=" = 3 } }
+occurrence_period = 
+#description = "Cough"
+
+[[observation]]
+name = "cough"
+date = { ref = "admissionDateHierarchy" } 
+phase = "study"
+is_present = { if field = "daily_cough_ceoccur_v2", ref = "Y/N/NK" }
+if = { daily_cough_ceoccur_v2 = { "!=" = 3 } }
+occurrence_period = 
+#description = "Cough"
+
+[[observation]]
+name = "cough"
+date = { ref = "admissionDateHierarchy" } 
+phase = "admission"
+is_present = { if field = "symptoms_cough", values = { 1 = true, 0 = false } }
+if = { symptoms_cough = { "!=" = "" } }
+occurrence_period = 
+#description = "Cough"
+
+[[observation]]
+name = "avpu"
+date = { ref = "admissionDateHierarchy" }
+phase = "study"
+value_char = { field = "avpu_vsorres", values = { 1 = "Alert", 2 = "Verbal", 3 = "Pain", 4 = "Unresponsive" }}
+if = { avpu_vsyn = 1 }
+
+[[observation]]
+name = "avpu"
+date = { ref = "admissionDateHierarchy" }
+phase = "study"
+value_char = { field = "daily_avpu_vsorres", values = { 1 = "Alert", 2 = "Verbal", 3 = "Pain", 4 = "Unresponsive" }}
+if = { daily_avpu_vsyn = 1 }
+
+[[observation]]
+name = "oxygen_saturation_percent"
+phase = "admission"
+date = { ref = "admissionDateHierarchy" }
+context = { field = "oxy_vsorresu", description = "Oxygen saturation on:", values = { 1 = ["Room air"], 2 = ["Oxygen therapy"]} }
+value_num = { field = "oxy_vsorres" }
+if = { oxy_vsyn = 1 }
+
+[[observation]]
+name = "oxygen_saturation_percent"
+phase = "admission"
+date = { ref = "admissionDateHierarchy" }
+context = { field = "daily_oxy_vsorresu", description = "Oxygen saturation on:", values = { 1 = ["Room air"], 2 = ["Oxygen therapy"]} }
```

Comment:
Same here, might be better to use firstNotNull for these

Reply:
we can discuss the format for this. Now I really miss json. It was much easier to write come to think of it.